include/xtensor/xstrided_view.hpp

```diff
@@ -807,6 +807,51 @@ namespace xt
         );
     }
 
+    namespace detail
+    {
+        // if shape is signed do the check
+        template <class S, class R>
+        using do_shape_recalculation = std::
+            enable_if_t<std::is_signed<typename std::decay<S>::type::value_type>::value, R>;
```

Comment:
You can use [get_value_type_t](https://github.com/xtensor-stack/xtensor/blob/5f49f6407501474cad622f9f16da172ab3717de3/include/xtensor/xutils.hpp#L496) to get the `value_type` of `S`. Notice that the implementatino does not decay `S`, so you might need to fix it.